sdk/core/azure-core/src/main/java/com/azure/core/util/implementation/BinaryDataContent.java

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util.implementation;
+
+import com.azure.core.util.BinaryData;
+import com.azure.core.util.serializer.JsonSerializer;
+import com.azure.core.util.serializer.JsonSerializerProviders;
+import com.azure.core.util.serializer.ObjectSerializer;
+import com.azure.core.util.serializer.TypeReference;
+import reactor.core.publisher.Flux;
+
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.nio.ReadOnlyBufferException;
+
+/**
+ * An abstract internal representation of the content stored in {@link BinaryData}.
+ */
+public abstract class BinaryDataContent {
+
+    static final byte[] ZERO_BYTE_ARRAY = new byte[0];
+    static final int STREAM_READ_SIZE = 1024;
+    static final JsonSerializer SERIALIZER = JsonSerializerProviders.createInstance(true);
```

Comment:
Does this class and `BinaryData` both need to have a singleton instance of the serializer? Instead could this class have a constructor which takes the serializer as a parameter?